flash/tabular/regression/data.py

```diff
@@ -0,0 +1,214 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from flash.core.data.callback import BaseDataFetcher
+from flash.core.data.data_module import DataModule
+from flash.core.data.process import Preprocess
+from flash.core.utilities.imports import _PANDAS_AVAILABLE
+
+if _PANDAS_AVAILABLE:
+    import pandas as pd
+    from pandas.core.frame import DataFrame
+else:
+    DataFrame = object
+
+from flash.tabular.data import TabularData
+
+
+class TabularRegressionData(TabularData):
+
+    @classmethod
+    def from_data_frame(

```

Comment:
This would enable us to duplicate their classmethod as they would live on the TabularData class.